src/App.tsx

```diff
@@ -1,26 +1,50 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect, useReducer } from 'react'
+import SearchPage from './components/SearchPage'
+import MyCollectionsPage from './components/MyCollectionsPage'
+import Route from './components/Route'
+import Link from './components/Link'
+import { collectionReducer, MyCollectionContext, fetchCards } from 'utils'
+
+const App = () => {
+  const [myCollection, dispatch] = useReducer(collectionReducer, {
+    cards: [],
+  })
+
+  //load an initial card
+  useEffect(() => {
+    fetchCards('https://api.pokemontcg.io/v2/cards?q=id:basep-1').then(
+      (result) => {
+        dispatch({
+          type: 'ADD-CARD',
+          card: result.data[0],
+        })
+      }
+    )
+  }, [])
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    <MyCollectionContext.Provider value={{ myCollection, dispatch }}>
```

Comment:
I'd use a custom provider instead of directly using context: https://kentcdodds.com/blog/how-to-use-react-context-effectively